src/Compilers/CSharp/Portable/Syntax/SyntaxFactory.cs

```diff
@@ -1876,15 +1876,21 @@ public static BracketedArgumentListSyntax ParseBracketedArgumentList(string text
         /// <param name="options">The optional parse options to use. If no options are specified default options are
         /// used.</param>
         /// <param name="consumeFullText">True if extra tokens in the input should be treated as an error</param>
-        public static AttributeArgumentListSyntax ParseAttributeArgumentList(string text, int offset = 0, ParseOptions? options = null, bool consumeFullText = true)
-        {
-            using (var lexer = MakeLexer(text, offset, (CSharpParseOptions?)options))
-            using (var parser = MakeParser(lexer))
-            {
-                var node = parser.ParseAttributeArgumentList();
```

Comment:
this code would crash before, and NRT revealed it.  ParseAttributeArgumentList can return null, and this code was not resilient to that.  test added.